lib/calculate_reference_proteome_similarity.py

```diff
@@ -1,21 +1,30 @@
 import csv
 from Bio.Blast import NCBIWWW
 from Bio.Blast import NCBIXML
-from Bio import SeqIO
+from Bio import SeqIO, SearchIO
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
+from Bio.Alphabet import IUPAC
 import shutil
 import re
 import os
 from collections import defaultdict
+from subprocess import run, DEVNULL, STDOUT
+import tempfile
 
 class CalculateReferenceProteomeSimilarity:
-    def __init__(self, input_file, input_fasta, output_file, match_length=8, species='human', file_type='pVACseq'):
+    def __init__(self, input_file, input_fasta, output_file, match_length=8, species='human', file_type='pVACseq', blastp_path = None, blastp_db = 'refseq_select_prot'):
         self.input_file = input_file
         self.input_fasta = input_fasta
         self.output_file = output_file
         self.metric_file = "{}.reference_matches".format(output_file)
         self.match_length = match_length
         self.species = species
         self.file_type = file_type
+        self.blastp_path = blastp_path
+        self.blastp_db = blastp_db
+        if self.blastp_db == 'refseq_select_prot' and self.species != 'human' and self.species != 'mouse':
+            raise Exception("refseq_select_prot blastp database is only compatible with human and mouse species.")
```

Comment:
Why is this restriction in place? Shouldn't the blastp algorithm be able to handle any species a user cares to build a database for?

Reply:
The `refseq_select_prot` database specifically is a limited set of reference proteins and only includes human and mouse. If someone wants to run other species they need to install the full reference proteome database (the `refseq_protein` option). I chose `refseq_select_prot` as the default since it will be faster than the full database for users running with the API.

Reply:
See https://ncbiinsights.ncbi.nlm.nih.gov/2020/10/16/blast-refseq-select/ for more info on the refseq select database.